### PR TITLE
Adding graham.c

### PR DESCRIPTION
**Pull request by Gathros**

### CODE REVIEW

File: chapters/computational_geometry/gift_wrapping/graham_scan/code/c/graham.c [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+struct point {
+    double x, y;
+};
+
+int cmp_points(const void *a, const void *b) {
+    struct point point1 = *(struct point*)a;
```
Comment (Gustorn) [inline chapters/computational_geometry/gift_wrapping/graham_scan/code/c/graham.c]: You shouldn't copy the points, it's completely unnecessary. Just cast to `struct point*` and compare the points that way

File: chapters/computational_geometry/gift_wrapping/graham_scan/code/c/graham.c [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    if (point1.y > point2.y) {
+        return 1;
+    } else if (point1.y < point2.y) {
+        return -1;
+    } else {
+        return 0;
+    }
+}
+
+double ccw(struct point a, struct point b, struct point c) {
```
Comment (Gustorn) [inline chapters/computational_geometry/gift_wrapping/graham_scan/code/c/graham.c]: I don't think `ccw` is the best name here. That would make sense if it returned a `bool`. I'd probably call it `orientation` or something like that.

**Comment by Gathros** [comment]
@Gustorn I changed `ccw` to `is_left_of` and changed `cmp_points`.

File: chapters/computational_geometry/gift_wrapping/graham_scan/code/c/graham.c [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+struct point {
+    double x, y;
+};
+
+int cmp_points(const void *a, const void *b) {
+    if (((struct point*)a)->y > ((struct point*)b)->y) {
```
Comment (Gustorn) [inline chapters/computational_geometry/gift_wrapping/graham_scan/code/c/graham.c]: Having them as variables wasn't a bad idea, I just didn't like the copy:

```c
struct point* pa = (struct point*) a;
struct point* pb = (struct point*) b;

if (pa->y > pb->y) {
// ...
```

**Comment by julianschacherpp** [comment]
Since @Gustorn approved it, it should be fine :)

**Pull request merged**